posthog/temporal/common/clickhouse.py

```diff
@@ -49,12 +49,29 @@ def encode_clickhouse_data(data: typing.Any) -> bytes:
             return result
 
         case dict():
-            return json.dumps(data).encode("utf-8")
+            # Assume we are working with JSON data.
+            encoded_data = []
+
+            for key, value in data.items():
+                if isinstance(value, dt.datetime):
+                    value = value.timestamp()
+                elif isinstance(value, uuid.UUID) or isinstance(value, str):
+                    value = str(value)
+                    quote_char = '"'  # JSON requires double quotes.
+                else:
+                    quote_char = "'"
+
+                encoded_data.append(
+                    f'"{str(key)}"'.encode("utf-8") + b":" + encode_clickhouse_data(value, quote_char=quote_char)
+                )
+
+            result = b"{" + b",".join(encoded_data) + b"}"
+            return result
```

Comment:
A comment here about why we're building these strings this way would be helpful (it's not obvious to me, at least.)

Reply:
I've extended the top comment explaining a bit more why we do JSON encoding for dictionaries and its limitations.